From 287dbc6ee59efe6d49a886582afbded0a5eaa7a8 Mon Sep 17 00:00:00 2001
From: from minimyth2 <unknown@unknown.org>
Date: Thu, 18 Jan 2024 08:23:36 +0100
Subject: [PATCH 119/184] 
 0565-arm64-dts-allwinner-tanix-tx6-mini-enable-wifi-cpu-dvfs.patch

---
 .../allwinner/sun50i-h6-tanix-tx6-mini.dts    | 47 +++++++++++++++++++
 1 file changed, 47 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts b/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts
index d9119e1cd..6c86a2298 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts
@@ -4,6 +4,7 @@
 /dts-v1/;
 
 #include "sun50i-h6-tanix.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 
 / {
 	model = "Tanix TX6 mini";
@@ -11,10 +12,37 @@ / {
 
 	aliases {
 		ethernet0 = &emac;
+		ethernet1 = &xr819;
 	};
 
+	reg_vcc33_wifi: vcc33-wifi {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc33-wifi";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&pio 7 4 GPIO_ACTIVE_HIGH>; /* PH4 */
+		regulator-always-on;
+		enable-active-high;
+	};
+
+	reg_vdd_cpu: vdd-cpu {
+		compatible = "regulator-fixed";
+		regulator-name = "vdd-cpu";
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+	};
 };
 
+&cpu0 {
+	cpu-supply = <&reg_vdd_cpu>;
+};
+
+&wifi_pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		reset-gpios = <&r_pio 1 3 GPIO_ACTIVE_LOW>; /* PM3 */
+		post-power-on-delay-ms = <200>;
+	};
+
 &r_ir {
 	linux,rc-map-name = "rc-tanix-tx3mini";
 };
@@ -45,3 +73,22 @@ ext_rmii_phy: ethernet-phy@1 {
 &pwm {
 	status = "okay";
 };
+
+&mmc1 {
+	vmmc-supply = <&reg_vcc3v3>;
+	vqmmc-supply = <&reg_vcc33_wifi>;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	bus-width = <4>;
+	non-removable;
+	status = "okay";
+
+	xr819: sdio_wifi@1 {
+		reg = <1>;
+		compatible = "xradio,xr819";
+		interrupt-parent = <&r_pio>;
+		interrupts = <1 0 IRQ_TYPE_EDGE_RISING>; /* PM0 */
+		interrupt-names = "host-wake";
+		local-mac-address = [dc 44 6d c0 ff 01];
+		keep-power-in-suspend;
+	};
+};
-- 
2.39.2

